Polish up provisioners.mdx (#236)
smallstep/docs: step-ca/provisioners.mdx

GCP    | ✔️  | ✔️  | 𝗫 | 𝗫 | ✔️  | 𝗫 | 𝗫 | 𝗫 | 𝗫

<br></br>

**Notes:**
<Footnote id="f1" marker="1"> 
By design, SSH user certificates cannot be renewed by any provisioner. 
However, you can re-use the same key to get a new certificate. 
See [smallstep/discussions#1296](https://github.com/smallstep/certificates/discussions/1296).
</Footnote>
<Footnote id="f2" marker="2"> 
Privileged OIDC subject names can generate Host SSH Certificates, 
and can be configured in the <a href="#oauthoidc-single-sign-on">OIDC provisioner</a>.
</Footnote>

To illustrate how to interpret this table, let's take the `JWK` and `SSHPOP` provisioner. 

A `JWK` provisioner _can_ sign, renew, and revoke X.509 certificates, as well as sign user and host SSH certificates. 

> However, it _cannot_ renew or rekey SSH certificates.

Reviewer: ```suggestion
However, it _cannot_ renew SSH user certificates.<Reference id="f1" marker="1"/>
```

On the other hand, an `SSHPOP` provisioner _can_ revoke and rekey SSH certificates 
and renew SSH host certificates. 
However, it _cannot_ sign, renew, or revoke X.509 certificates. 
Additionally, it _cannot_ sign SSH user and host certificates or renew SSH user certificates. 

It's important to understand the capabilities and limitations when selecting a 
provisioner for a given workload.

## Provisioner Management

> To add, remove, or modify provisioner configurations, 

Reviewer: I'd suggest changing this section so that it begins by explaining that provisioner configuration can reside either in `ca.json`, or in the database. It resides in `ca.json` by default. Database-stored provisioners are good if you have multiple load-balanced CA nodes. Refer them to the [Remote Provisioner Management](https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management) section for more about the tradeoffs between these two options.

The `step ca provisioner` commands try to detect where the provisioner configuration is stored. It will modify the `ca.json` file directly, or it will update the database, depending on whether you have enabled [Remote Provisioner Management](https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management).

Author: Novice question: I've seen the phrase 'the database' across our docs. But I never quite understood what database is being referred to. Where is 'this' database emanates from? The term 'the database' insinuates that there is a singular database. Is this the case?

Author: Also, just noticed that the provisioner management section doesn't exactly discuss the tradeoffs or differences between the default and remote provisioner management options.

Author: I made changes in this pull request https://github.com/smallstep/docs/pull/239

You can authenticate as an administrator via the authentication scheme of the provisioner.
For example, for the default JWK provisioner, running administrative commands in `step` will prompt for the provisioner password.

> ***Tip: Single Sign-On (SS0) can be used for Admin users.*** 

Reviewer: ```suggestion
***Tip: Single Sign-On (SSO) can be used for Admin users.*** 
```

### SCEP

The SCEP provisioner can sign and renew certificates 
using the Simple Certificate Enrollment Protocol SCEP protocol (SCEP) ([RFC8894](https://datatracker.ietf.org/doc/html/rfc8894)). 

> SCEP is popularly used for certificate management, 
> particularly in network equipment and mobile device management (MDM). 

Reviewer: ```suggestion
SCEP is commonly used for certificate management 
in network equipment and mobile device management (MDM). 
```
